Remove lint warning about unexpected indentation outside of indent rule

paul-dingemans opened the pull request:
## Description

Only the indentation rule should emit warnings about incorrect indentation to avoid
conflicting warnings from different rules about the indentation of the exact same
line. However, those other rules should still fix the indentation as good as they
can for the case that the indent rule is not run at all.

Closes #1267, #1119, #1045, #1255, #1319, #1320, #897, #1337

## Checklist

<!-- Following checklist maybe skipped in some cases -->
- [ ] tests are added
- [X] `CHANGELOG.md` is updated

shashachu commented on ktlint-ruleset-experimental/src/main/kotlin/com/pinterest/ktlint/ruleset/experimental/ArgumentListWrappingRule.kt:
```diff
@@ -139,18 +139,19 @@ class ArgumentListWrappingRule : Rule("argument-list-wrapping") {
                                 if (autoCorrect) {
-                                    val finalIndent =
-                                        (if (cut > -1) spacing.substring(0, cut) else "") + intendedIndent
+                                    // Although no warning is displayed about incorrect indentation, an attempt is
```
This seems strange that we wouldn't emit a warning on linting, but we'd change indentation on formatting.

paul-dingemans replied:
Yes it is a bit unusual. But look at the number of issue we get about those warnings. So the warning has to be removed at minimum. Keeping the indent fix during format only has value in case the indent rule is disabled.

shashachu replied:
We shouldn't have different behavior for linting vs formatting.

paul-dingemans replied:
I am fine with removing the functionality to fix indentation inside the argument/parameter list wrapping rule in order to make linting and formatting consistent again. Some users might consider this a breaking change.

@shashachu please confirm whether that is the way to go.

romtsn replied:
yes, let's disable autocorrection of indentation for now

paul-dingemans commented:
Like before a warning is emitted in case the current indentation is not as expected. However, the emitted warning now no longer reports the expected indentation but just that the argument/parameter should be on a separate line unless all of them fit on a single line.

romtsn commented:
hm, can we remove autocorrection of indentation entirely from these rules? let's just delegate this to the indent-rule completely and say that we do not support indentation alignment in these wrapping rules (even the names suggest that they are wrapping and not indenting :))

paul-dingemans commented:
> hm, can we remove autocorrection of indentation entirely from these rules? let's just delegate this to the indent-rule completely and say that we do not support indentation alignment in these wrapping rules (even the names suggest that they are wrapping and not indenting :))

I think this would give a bad user experience in case the indent rule would be disabled. More importantly is that it would have a big Ktlint-developer impact as this would also imply that the expected formatted code in unit tests would have to be de-indented. So, writing a test like below:
```
    @Test
    fun testFormatArgumentsWithNestedCalls() {
        assertThat(
            ArgumentListWrappingRule().format(
                """
                val x = test(
                    one("a", "b",
                    "c"),
                    "Two", "Three", "Four"
                )
                """.trimIndent()
            )
        ).isEqualTo(
            """
            one(
            "a",
            "b",
            "c"
            ),
            "Two",
            "Three",
            "Four"
            )
            """.trimIndent()
        )
    }
```

romtsn commented:
gotcha, let's merge it like that then!

paul-dingemans commented:
When checking all reported issues, I encountered a problem which I did not see before. The incorrect indentation warning was sometimes replace with another incorrect warning. No I have changed the rules to ignore the indentation if the the whitespace before the parameter, argument or closing parathesis contains a new line because this means that the wrapping was done correctly. Previously an incorrect warning was shown in case that whitespace contained a newline but was incorrectly indented according to that rule.

In order to keep the unit tests of the rules readable i have modified the lint/format utilities to accept a list of rules instead of a single rule. In this way, I can execute both the wrapping as well as the indentation rule together. I am going to need that functionality also for the new signature rewrite rule that I am developing, so I think it is a good addition to our testing toolkit.

I have, once more checked that all issues associated with this PR are now indedd resolved. If the build succeeds, I will merge to master.
